docs/20_smart-contracts/10_getting-started/10_dune-guide/index.md

```diff
@@ -0,0 +1,634 @@
+      };
+      typedef eosio::multi_index< name("users"), user_record> user_index;
+
+      ACTION world( name user ) {
+         user_index users( get_self(), get_self().value );
+         users.emplace( get_self(), [&]( auto& new_record ) {
+            new_record.user = user;
+         });
+         print_f("Hello World from %!\n", user);
+      }
+};
+```
+
+### Recompile, Deploy, Test
+
+Now that you have added state to your `hello` contract by creating a multi-index table, you can compile, deploy, and test your modified contract. To that end, re-execute the following commands:
+
+```shell
+dune --cmake-build .
+```
+```
+-- Configuring done
+-- Generating done
+-- Build files have been written to: /host/Users/lparisc/Documents/eosnf/work/contracts/hello/build
+[ 11%] Performing build step for 'hello_project'
+Scanning dependencies of target hello
+[ 50%] Building CXX object CMakeFiles/hello.dir/hello.obj
+Warning, empty ricardian clause file
+[100%] Linking CXX executable hello.wasm
+[100%] Built target hello
+[ 22%] No install step for 'hello_project'
+[ 33%] No test step for 'hello_project'
+[ 44%] Completed 'hello_project'
+[100%] Built target hello_project
+```
+
+```shell
+dune --deploy build/hello hello
+```
+```
+#         eosio <= eosio::setcode               {"account":"hello","vmtype":0,"vmversion":0,"code":"0061736d010000000159106000006000017f60027f7f0060...
+#         eosio <= eosio::setabi                {"account":"hello","abi":"0e656f73696f3a3a6162692f312e3200020b757365725f7265636f72640001047573657204...
+```
+
+If all went well, you can now send new `world` actions to your `hello` contract by passing different names - just make sure they are unique and they abide by the EOS name rules:
+
+```shell
+dune --send-action hello world '[bob]' hello@active
+```
+```
+executed transaction: d32628565fee76440da0a8bc8d2f0eacce15dbdd629b3a64e8aad21ae87171eb  104 bytes  100 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ]
+#         hello <= hello::world                 {"user":"bob"}
+>> Hello World from bob!
+```
+
+```shell
+dune --send-action hello world '[alice]' hello@active
+```
+```
+executed transaction: c439721576b38f68714a606fbfd150e7a3e870bd71361804b7098a3aabce5eeb  104 bytes  114 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ]
+#         hello <= hello::world                 {"user":"alice"}
+>> Hello World from alice!
+```
+
+```shell
+dune --send-action hello world '[john]' hello@active
+```
+```
+executed transaction: ff05d2842ae5d6def42d96474837ca03d5106f6f94ce62f2a579905143376ab7  104 bytes  103 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ]
+#         hello <= hello::world                 {"user":"john"}
+>> Hello World from john!
+```
+
+Note that if you try to re-add the same user name, the transaction will fail as expected:
```

Comment:
Like the comment above, we don't want to teach them about the contents of the smart contract (assertions), just the concepts of sending actions and retrieving data using DUNE

Reply:
Removed and simply mentioned the action will fail if user is added twice.